debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -82,7 +82,29 @@ public class JdbcConnection implements AutoCloseable {
 
     private static final int WAIT_FOR_CLOSE_SECONDS = 10;
     private static final char STATEMENT_DELIMITER = ';';
-    private static final String ESCAPE_CHAR = "\\";
+
+    /**
+     * The escape character used to escape special characters in the patterns being passed to the JDBC's
+     * DatabaseMetaData methods.
+     */
+    private static final Character ESCAPE_CHAR = '\\';
+
+    /**
+     * The special characters that must be escaped when constructing patterns for JDBC {@link java.sql.DatabaseMetaData}
+     * methods such as {@code getColumns}, {@code getTables}, etc.
+     *
+     * <p>These characters require escaping due to their special meaning in SQL {@code LIKE} patterns:
+     *
+     * <ol>
+     *     <li>{@code %} and {@code _} are wildcard characters defined by the ANSI SQL standard.</li>
+     *     <li>{@code [} is used in SQL Server's extended pattern syntax to define character ranges or sets.</li>
+     *     <li>The escape character itself (denoted by {@code ESCAPE_CHAR}) must also be escaped.</li>
+     * </ol>
+     *
+     * @see <a href="https://learn.microsoft.com/en-us/sql/t-sql/language-elements/like-transact-sql#pattern">SQL Server LIKE pattern syntax</a>
+     */
+    private static final Set<Character> SPECIAL_CHARS = Set.of('%', '_', '[', ESCAPE_CHAR);
```

Comment:
Technically, the SQL Server specific character doesn't belong to the generic `JdbcConnection` class. But this is so nice an cozy to have everything defined in a single `private static final` field. We can take it apart, but then we'd have to make it non-static and protected and initialize/override in constructors.